packit/api.py

```diff
@@ -269,6 +269,37 @@ def pkg_tool(self) -> str:
 
         return self.config.pkg_tool
 
+    def _get_common_env_paths(self) -> dict[str, str]:
+        paths = {
+            variable_name: str(repo.local_project.working_dir)
+            for variable_name, repo in (
+                ("PACKIT_DOWNSTREAM_REPO", self.dg),
+                ("PACKIT_UPSTREAM_REPO", self.up),
+                ("PACKIT_PWD", self.up),
+            )
+            if isinstance(repo, PackitRepositoryBase)
+            and repo._local_project is not None
+        }
+        if isinstance(self.up, NonGitUpstream):
+            paths.update({"PACKIT_PWD": str(self.up.working_dir)})
+
+        # Adjust paths for the sandcastle
+        if self.config.command_handler == RunCommandType.sandcastle:
```

Comment:
Should this not be generalised to a function? Like a normalize path function.

Doesn't `PACKIT_SPECFILE_PATH` need the same handling?